diff --git a/src/core/credentials/transport/ssl/ssl_credentials.cc b/src/core/credentials/transport/ssl/ssl_credentials.cc
index 8d6411ed..82ae022a 100644
--- a/src/core/credentials/transport/ssl/ssl_credentials.cc
+++ b/src/core/credentials/transport/ssl/ssl_credentials.cc
@@ -221,11 +221,6 @@ grpc_security_status grpc_ssl_credentials::InitializeClientHandshakerFactory(
 grpc_channel_credentials* grpc_ssl_credentials_create(
     const char* pem_root_certs, grpc_ssl_pem_key_cert_pair* pem_key_cert_pair,
     const verify_peer_options* verify_options, void* reserved) {
-  GRPC_TRACE_LOG(api, INFO)
-      << "grpc_ssl_credentials_create(pem_root_certs=" << pem_root_certs
-      << ", pem_key_cert_pair=" << pem_key_cert_pair
-      << ", verify_options=" << verify_options << ", reserved=" << reserved
-      << ")";
   GRPC_CHECK_EQ(reserved, nullptr);
 
   return new grpc_ssl_credentials(
@@ -236,11 +231,6 @@ grpc_channel_credentials* grpc_ssl_credentials_create(
 grpc_channel_credentials* grpc_ssl_credentials_create_ex(
     const char* pem_root_certs, grpc_ssl_pem_key_cert_pair* pem_key_cert_pair,
     const grpc_ssl_verify_peer_options* verify_options, void* reserved) {
-  GRPC_TRACE_LOG(api, INFO)
-      << "grpc_ssl_credentials_create(pem_root_certs=" << pem_root_certs
-      << ", pem_key_cert_pair=" << pem_key_cert_pair
-      << ", verify_options=" << verify_options << ", reserved=" << reserved
-      << ")";
   GRPC_CHECK_EQ(reserved, nullptr);
 
   return new grpc_ssl_credentials(pem_root_certs, pem_key_cert_pair,
@@ -419,12 +409,6 @@ grpc_server_credentials* grpc_ssl_server_credentials_create_ex(
     size_t num_key_cert_pairs,
     grpc_ssl_client_certificate_request_type client_certificate_request,
     void* reserved) {
-  GRPC_TRACE_LOG(api, INFO)
-      << "grpc_ssl_server_credentials_create_ex(pem_root_certs="
-      << pem_root_certs << ", pem_key_cert_pairs=" << pem_key_cert_pairs
-      << ", num_key_cert_pairs=" << (unsigned long)num_key_cert_pairs
-      << ", client_certificate_request=" << client_certificate_request
-      << ", reserved=" << reserved << ")";
   GRPC_CHECK_EQ(reserved, nullptr);
 
   grpc_ssl_server_certificate_config* cert_config =
diff --git a/src/core/transport/auth_context.cc b/src/core/transport/auth_context.cc
index 026742f1..20bac288 100644
--- a/src/core/transport/auth_context.cc
+++ b/src/core/transport/auth_context.cc
@@ -158,11 +158,6 @@ void grpc_auth_context::set_protocol(absl::string_view protocol) {
 
 void grpc_auth_context_add_property(grpc_auth_context* ctx, const char* name,
                                     const char* value, size_t value_length) {
-  GRPC_TRACE_LOG(api, INFO) << absl::StrFormat(
-      "grpc_auth_context_add_property(ctx=%p, name=%s, value=%*.*s, "
-      "value_length=%lu)",
-      ctx, name, (int)value_length, (int)value_length, value,
-      (unsigned long)value_length);
   ctx->add_property(name, value, value_length);
 }
 
