diff --git a/src/3rdparty/chromium/build/config/compiler/BUILD.gn b/src/3rdparty/chromium/build/config/compiler/BUILD.gn
index b511a58c5..0e681d58a 100644
--- a/src/3rdparty/chromium/build/config/compiler/BUILD.gn
+++ b/src/3rdparty/chromium/build/config/compiler/BUILD.gn
@@ -1284,9 +1284,6 @@ config("default_warnings") {
     if (treat_warnings_as_errors) {
       cflags += [ "/WX" ]
     }
-    if (fatal_linker_warnings) {
-      ldflags = [ "/WX" ]
-    }
 
     cflags += [
       # Warnings permanently disabled:
diff --git a/src/3rdparty/chromium/ui/accessibility/ax_table_info.h b/src/3rdparty/chromium/ui/accessibility/ax_table_info.h
index 3bfb24626..cc3d51b27 100644
--- a/src/3rdparty/chromium/ui/accessibility/ax_table_info.h
+++ b/src/3rdparty/chromium/ui/accessibility/ax_table_info.h
@@ -9,6 +9,7 @@
 #include <set>
 #include <unordered_map>
 #include <vector>
+#include <string>
 
 #include "base/optional.h"
 #include "ui/accessibility/ax_export.h"
