diff --git a/include/boost/test/impl/framework.ipp b/include/boost/test/impl/framework.ipp
index b139e8db..c7c35388 100644
--- a/include/boost/test/impl/framework.ipp
+++ b/include/boost/test/impl/framework.ipp
@@ -1195,7 +1195,7 @@ init( init_unit_test_func init_func, int argc, char* argv[] )
     // 50. Set up memory leak detection
     unsigned long detect_mem_leak = runtime_config::get<unsigned long>( runtime_config::btrt_detect_mem_leaks );
     if( detect_mem_leak > 0 ) {
-        debug::detect_memory_leaks( true, runtime_config::get<std::string>( runtime_config::btrt_report_mem_leaks ) );
+        debug::detect_memory_leaks( false, runtime_config::get<std::string>( runtime_config::btrt_report_mem_leaks ) );
         debug::break_memory_alloc( (long)detect_mem_leak );
     }
 
