diff --git a/src/options.c b/src/options.c
index 9da3a9b..03bdde9 100644
--- a/src/options.c
+++ b/src/options.c
@@ -66,7 +66,7 @@ static mi_option_desc_t options[_mi_option_last] =
 
   // the following options are experimental and not all combinations make sense.
   { 1, UNINIT, MI_OPTION(eager_commit) },        // commit per segment directly (4MiB)  (but see also `eager_commit_delay`)
-  #if defined(_WIN32) || (MI_INTPTR_SIZE <= 4)   // and other OS's without overcommit?
+  #if MI_INTPTR_SIZE <= 4   // and other OS's without overcommit?
   { 0, UNINIT, MI_OPTION(eager_region_commit) },
   { 1, UNINIT, MI_OPTION(reset_decommits) },     // reset decommits memory
   #else
