diff --git a/include/mimalloc-stats.h b/include/mimalloc-stats.h
index 44c4886..3aa48bb 100644
--- a/include/mimalloc-stats.h
+++ b/include/mimalloc-stats.h
@@ -37,6 +37,7 @@ typedef struct mi_stat_counter_s {
   MI_STAT_COUNT(malloc_normal)              /* allocated bytes <= MI_LARGE_OBJ_SIZE_MAX */ \
   MI_STAT_COUNT(malloc_huge)                /* allocated bytes in huge pages */ \
   MI_STAT_COUNT(malloc_requested)           /* malloc requested bytes */ \
+  MI_STAT_COUNT(prqa_malloc_requested)      /* malloc requested bytes */ \
   \
   MI_STAT_COUNTER(mmap_calls) \
   MI_STAT_COUNTER(commit_calls) \
diff --git a/include/mimalloc.h b/include/mimalloc.h
index ae6ae26..c0ad687 100644
--- a/include/mimalloc.h
+++ b/include/mimalloc.h
@@ -103,6 +103,19 @@ terms of the MIT license. A copy of the license can be found in the file
 extern "C" {
 #endif
 
+// ------------------------------------------------------
+// PRQA interface
+// ------------------------------------------------------
+
+typedef struct mi_prqa_stat_s {
+    int64_t current;
+    int64_t peak;
+} mi_prqa_stat_t;
+
+mi_decl_nodiscard mi_decl_export mi_prqa_stat_t mi_get_prqa_malloc_stat(void) mi_attr_noexcept;
+mi_decl_export void mi_reset_prqa_malloc_stat(void) mi_attr_noexcept;
+mi_decl_export void mi_track_prqa_malloc_stat(bool enable) mi_attr_noexcept;
+
 // ------------------------------------------------------
 // Standard malloc interface
 // ------------------------------------------------------
@@ -155,6 +168,7 @@ mi_decl_export void mi_stats_merge(void)      mi_attr_noexcept;
 mi_decl_export void mi_stats_print(void* out) mi_attr_noexcept;  // backward compatibility: `out` is ignored and should be NULL
 mi_decl_export void mi_stats_print_out(mi_output_fun* out, void* arg) mi_attr_noexcept;
 mi_decl_export void mi_options_print(void)    mi_attr_noexcept;
+mi_decl_export void mi_pretty_print_bytes(int64_t n, char* out, int len) mi_attr_noexcept;
 
 mi_decl_export void mi_process_init(void)     mi_attr_noexcept;
 mi_decl_export void mi_thread_init(void)      mi_attr_noexcept;
diff --git a/include/mimalloc/internal.h b/include/mimalloc/internal.h
index c53c841..6eca975 100644
--- a/include/mimalloc/internal.h
+++ b/include/mimalloc/internal.h
@@ -236,6 +236,22 @@ mi_msecs_t  _mi_clock_now(void);
 mi_msecs_t  _mi_clock_end(mi_msecs_t start);
 mi_msecs_t  _mi_clock_start(void);
 
+static inline void _mi_prqa_stat_increase(mi_stat_count_t* stat, size_t amount) {
+  if (amount) {
+    stat->current += (int64_t)amount;
+    if (stat->current > stat->peak) {
+      stat->peak = stat->current;
+    }
+  }
+}
+
+static inline void _mi_prqa_stat_decrease(mi_stat_count_t* stat, size_t amount) {
+  if (amount) {
+    stat->current -= (int64_t)amount;
+    mi_assert(stat->current >= 0);
+  }
+}
+
 // "alloc.c"
 void*       _mi_page_malloc_zero(mi_heap_t* heap, mi_page_t* page, size_t size, bool zero) mi_attr_noexcept;  // called from `_mi_malloc_generic`
 void*       _mi_page_malloc(mi_heap_t* heap, mi_page_t* page, size_t size) mi_attr_noexcept;                  // called from `_mi_heap_malloc_aligned`
@@ -1100,5 +1116,10 @@ static inline void _mi_memzero_aligned(void* dst, size_t n) {
 }
 #endif
 
+// -------------------------------------------------------------------------------
+// Statistics
+// -------------------------------------------------------------------------------
+
+extern mi_decl_thread bool tls_prqa_stat;
 
 #endif
diff --git a/include/mimalloc/types.h b/include/mimalloc/types.h
index 1f4adc0..02ab1cb 100644
--- a/include/mimalloc/types.h
+++ b/include/mimalloc/types.h
@@ -700,5 +700,12 @@ void _mi_stat_counter_increase(mi_stat_counter_t* stat, size_t amount);
 #define mi_heap_stat_increase(heap,stat,amount)  mi_stat_increase( (heap)->tld->stats.stat, amount)
 #define mi_heap_stat_decrease(heap,stat,amount)  mi_stat_decrease( (heap)->tld->stats.stat, amount)
 
+// Applicable to release and debug builds
+
+#define mi_prqa_stat_increase(stat, amount)         _mi_prqa_stat_increase(&(stat), amount)
+#define mi_prqa_stat_decrease(stat, amount)         _mi_prqa_stat_decrease(&(stat), amount)
+
+#define mi_heap_prqa_stat_increase(heap, stat, amount)  mi_prqa_stat_increase((heap)->tld->stats.stat, amount)
+#define mi_heap_prqa_stat_decrease(heap, stat, amount)  mi_prqa_stat_decrease((heap)->tld->stats.stat, amount)
 
 #endif
diff --git a/src/alloc-aligned.c b/src/alloc-aligned.c
index bb9e9ec..876de4e 100644
--- a/src/alloc-aligned.c
+++ b/src/alloc-aligned.c
@@ -194,6 +194,9 @@ static void* mi_heap_malloc_zero_aligned_at(mi_heap_t* const heap, const size_t
         #if MI_STAT>1
         mi_heap_stat_increase(heap, malloc_requested, size);
         #endif
+        if (tls_prqa_stat) {
+          mi_heap_prqa_stat_increase(heap, prqa_malloc_requested, size);
+        }
         void* p = (zero ? _mi_page_malloc_zeroed(heap,page,padsize) : _mi_page_malloc(heap,page,padsize)); // call specific page malloc for better codegen
         mi_assert_internal(p != NULL);
         mi_assert_internal(((uintptr_t)p + offset) % alignment == 0);
diff --git a/src/alloc.c b/src/alloc.c
index 2350dc9..43f2ad4 100644
--- a/src/alloc.c
+++ b/src/alloc.c
@@ -150,8 +150,17 @@ static inline mi_decl_restrict void* mi_heap_malloc_small_zero(mi_heap_t* heap,
   if (p != NULL) {
     if (!mi_heap_is_initialized(heap)) { heap = mi_prim_get_default_heap(); }
     mi_heap_stat_increase(heap, malloc_requested, mi_usable_size(p));
+    if (tls_prqa_stat) {
+      mi_heap_prqa_stat_increase(heap, prqa_malloc_requested, mi_usable_size(p));
+    }
+  }
+  #else
+  if (p != NULL && tls_prqa_stat) {
+    if (!mi_heap_is_initialized(heap)) { heap = mi_prim_get_default_heap(); }
+    mi_heap_prqa_stat_increase(heap, prqa_malloc_requested, mi_usable_size(p));
   }
   #endif
+
   #if MI_DEBUG>3
   if (p != NULL && zero) {
     mi_assert_expensive(mi_mem_is_zero(p, size));
@@ -192,8 +201,17 @@ extern inline void* _mi_heap_malloc_zero_ex(mi_heap_t* heap, size_t size, bool z
     if (p != NULL) {
       if (!mi_heap_is_initialized(heap)) { heap = mi_prim_get_default_heap(); }
       mi_heap_stat_increase(heap, malloc_requested, mi_usable_size(p));
+      if (tls_prqa_stat) {
+        mi_heap_prqa_stat_increase(heap, prqa_malloc_requested, mi_usable_size(p));
+      }
+    }
+    #else
+    if (p != NULL && tls_prqa_stat) {
+      if (!mi_heap_is_initialized(heap)) { heap = mi_prim_get_default_heap(); }
+      mi_heap_prqa_stat_increase(heap, prqa_malloc_requested, mi_usable_size(p));
     }
     #endif
+
     #if MI_DEBUG>3
     if (p != NULL && zero) {
       mi_assert_expensive(mi_mem_is_zero(p, size));
diff --git a/src/free.c b/src/free.c
index 3aa6e36..e5592e8 100644
--- a/src/free.c
+++ b/src/free.c
@@ -520,17 +520,17 @@ static void mi_check_padding(const mi_page_t* page, const mi_block_t* block) {
 #endif
 
 // only maintain stats for smaller objects if requested
-#if (MI_STAT>0)
 static void mi_stat_free(const mi_page_t* page, const mi_block_t* block) {
-  #if (MI_STAT < 2)
-  MI_UNUSED(block);
-  #endif
+  #if (MI_STAT>0)
   mi_heap_t* const heap = mi_heap_get_default();
-  const size_t bsize = mi_page_usable_block_size(page);
-  #if (MI_STAT>1)
   const size_t usize = mi_page_usable_size_of(page, block);
-  mi_heap_stat_decrease(heap, malloc_requested, usize);
+  #if (MI_STAT>1)
+    mi_heap_stat_decrease(heap, malloc_requested, usize);
   #endif
+  if (tls_prqa_stat && _mi_thread_id() == heap->thread_id) {
+    mi_heap_prqa_stat_decrease(heap, prqa_malloc_requested, usize);
+  }
+  const size_t bsize = mi_page_usable_block_size(page);
   if (bsize <= MI_MEDIUM_OBJ_SIZE_MAX) {
     mi_heap_stat_decrease(heap, malloc_normal, bsize);
     #if (MI_STAT > 1)
@@ -543,13 +543,16 @@ static void mi_stat_free(const mi_page_t* page, const mi_block_t* block) {
   else {
     mi_heap_stat_decrease(heap, malloc_huge, bsize);
   }
+  #else
+  if (tls_prqa_stat) {
+    mi_heap_t* const heap = mi_heap_get_default();
+    if (_mi_thread_id() == heap->thread_id) {
+      const size_t usize = mi_page_usable_size_of(page, block);
+      mi_heap_prqa_stat_decrease(heap, prqa_malloc_requested, usize);
+    }
+  }
+  #endif
 }
-#else
-static void mi_stat_free(const mi_page_t* page, const mi_block_t* block) {
-  MI_UNUSED(page); MI_UNUSED(block);
-}
-#endif
-
 
 // Remove guard page when building with MI_GUARDED
 #if MI_GUARDED
diff --git a/src/heap.c b/src/heap.c
index bd3a1e2..38ede18 100644
--- a/src/heap.c
+++ b/src/heap.c
@@ -354,6 +354,15 @@ static bool _mi_heap_page_destroy(mi_heap_t* heap, mi_page_queue_t* pq, mi_page_
 #endif
   }
   mi_heap_stat_decrease(heap, malloc_requested, bsize * inuse);  // todo: off for aligned blocks...
+  if (tls_prqa_stat) {
+    mi_heap_prqa_stat_decrease(heap, prqa_malloc_requested, bsize * inuse);  // todo: off for aligned blocks...
+  }
+#else
+  if (tls_prqa_stat) {
+    _mi_page_free_collect(page, false);  // update used count
+    const size_t inuse = page->used;
+    mi_heap_prqa_stat_decrease(heap, prqa_malloc_requested, bsize * inuse);  // todo: off for aligned blocks...
+  }
 #endif
 
   /// pretend it is all free now
diff --git a/src/init.c b/src/init.c
index b0fb2e6..24fead8 100644
--- a/src/init.c
+++ b/src/init.c
@@ -73,7 +73,7 @@ const mi_page_t _mi_page_empty = {
 #define MI_STATS_NULL  \
   MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), \
   MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), \
-  MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), \
+  MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), MI_STAT_COUNT_NULL(), \
   { 0 }, { 0 }, { 0 }, { 0 }, \
   { 0 }, { 0 }, { 0 }, { 0 }, \
   \
diff --git a/src/stats.c b/src/stats.c
index d6c76fa..b4bb783 100644
--- a/src/stats.c
+++ b/src/stats.c
@@ -19,6 +19,29 @@ terms of the MIT license. A copy of the license can be found in the file
   Statistics operations
 ----------------------------------------------------------- */
 
+mi_decl_thread bool tls_prqa_stat = false;
+
+mi_decl_nodiscard mi_decl_export mi_prqa_stat_t mi_get_prqa_malloc_stat(void) mi_attr_noexcept {
+  mi_heap_t * heap = mi_prim_get_default_heap();
+  mi_assert(mi_heap_is_initialized(heap));
+  mi_stat_count_t * mc = &heap->tld->stats.prqa_malloc_requested;
+  mi_assert(mc->current >= 0);
+  mi_prqa_stat_t stat = { mc->current, mc->peak };
+  return stat;
+}
+
+mi_decl_export void mi_reset_prqa_malloc_stat(void) mi_attr_noexcept {
+  mi_heap_t * heap = mi_prim_get_default_heap();
+  mi_assert(mi_heap_is_initialized(heap));
+  mi_stat_count_t * mc = &heap->tld->stats.prqa_malloc_requested;
+  mc->current = 0;
+  mc->peak = 0;
+}
+
+mi_decl_export void mi_track_prqa_malloc_stat(bool enable) mi_attr_noexcept {
+  tls_prqa_stat = enable;
+}
+
 static bool mi_is_in_main(void* stat) {
   return ((uint8_t*)stat >= (uint8_t*)&_mi_stats_main
          && (uint8_t*)stat < ((uint8_t*)&_mi_stats_main + sizeof(mi_stats_t)));
@@ -105,12 +128,8 @@ static void mi_stats_add(mi_stats_t* stats, const mi_stats_t* src) {
   Display statistics
 ----------------------------------------------------------- */
 
-// unit > 0 : size in binary bytes
-// unit == 0: count as decimal
-// unit < 0 : count in binary
-static void mi_printf_amount(int64_t n, int64_t unit, mi_output_fun* out, void* arg, const char* fmt) {
-  char buf[32]; buf[0] = 0;
-  int  len = 32;
+static void mi_printf_amount_to_char_array(int64_t n, int64_t unit, char* buf, int len) {
+  buf[0] = 0;
   const char* suffix = (unit <= 0 ? " " : "B");
   const int64_t base = (unit == 0 ? 1000 : 1024);
   if (unit>0) n *= unit;
@@ -133,9 +152,16 @@ static void mi_printf_amount(int64_t n, int64_t unit, mi_output_fun* out, void*
     _mi_snprintf(unitdesc, 8, "%s%s%s", magnitude, (base==1024 ? "i" : ""), suffix);
     _mi_snprintf(buf, len, "%ld.%ld %-3s", whole, (frac1 < 0 ? -frac1 : frac1), unitdesc);
   }
-  _mi_fprintf(out, arg, (fmt==NULL ? "%12s" : fmt), buf);
 }
 
+// unit > 0 : size in binary bytes
+// unit == 0: count as decimal
+// unit < 0 : count in binary
+static void mi_printf_amount(int64_t n, int64_t unit, mi_output_fun* out, void* arg, const char* fmt) {
+  char buf[32];
+  mi_printf_amount_to_char_array(n, unit, buf, 32);
+  _mi_fprintf(out, arg, (fmt == NULL ? "%12s" : fmt), buf);
+}
 
 static void mi_print_amount(int64_t n, int64_t unit, mi_output_fun* out, void* arg) {
   mi_printf_amount(n,unit,out,arg,NULL);
@@ -146,6 +172,10 @@ static void mi_print_count(int64_t n, int64_t unit, mi_output_fun* out, void* ar
           else mi_print_amount(n,0,out,arg);
 }
 
+mi_decl_export void mi_pretty_print_bytes(int64_t n, char* out, int len) mi_attr_noexcept {
+  mi_printf_amount_to_char_array(n, 1, out, len);
+}
+
 static void mi_stat_print_ex(const mi_stat_count_t* stat, const char* msg, int64_t unit, mi_output_fun* out, void* arg, const char* notok ) {
   _mi_fprintf(out, arg,"%10s:", msg);
   if (unit != 0) {
