diff --git a/src/3rdparty/chromium/v8/src/snapshot/embedded/embedded-data.cc b/src/3rdparty/chromium/v8/src/snapshot/embedded/embedded-data.cc
index fb3642c959..02a21594e2 100644
--- a/src/3rdparty/chromium/v8/src/snapshot/embedded/embedded-data.cc
+++ b/src/3rdparty/chromium/v8/src/snapshot/embedded/embedded-data.cc
@@ -237,8 +237,8 @@ EmbeddedData EmbeddedData::NewFromIsolate(Isolate* isolate) {
   Builtins* builtins = isolate->builtins();
 
   // Store instruction stream lengths and offsets.
-  std::vector<struct LayoutDescription> layout_descriptions(kTableSize);
-  std::vector<struct BuiltinLookupEntry> offset_descriptions(kTableSize);
+  ::std::vector<struct ::v8::internal::EmbeddedData::LayoutDescription> layout_descriptions(kTableSize);
+  ::std::vector<struct ::v8::internal::EmbeddedData::BuiltinLookupEntry> offset_descriptions(kTableSize);
 
   bool saw_unsafe_builtin = false;
   uint32_t raw_code_size = 0;
