diff --git a/configure.cmake b/configure.cmake
index c0276751f..41745fcdd 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -601,6 +601,12 @@ if(WIN32)
                 CONDITION NOT MSVC_VERSION LESS 1936
                 MESSAGE "VS compiler version must be at least 14.36"
             )
+        elseif(MSVC_TOOLSET_VERSION EQUAL 145) # VS 2026 (18.0)
+            add_check_for_support(
+                MODULES QtWebEngine QtPdf
+                CONDITION NOT MSVC_VERSION LESS 1950
+                MESSAGE "VS compiler version must be at least 14.50"
+            )
         else()
             message(FATAL_ERROR "Build requires Visual Studio 2019 or higher.")
         endif()
