diff --git a/lib/zip_close.c b/lib/zip_close.c
index 6948c0e2..9c15c176 100644
--- a/lib/zip_close.c
+++ b/lib/zip_close.c
@@ -306,6 +306,7 @@ static int add_data(zip_t *za, zip_source_t *src, zip_dirent_t *de, zip_uint32_t
     bool needs_recompress, needs_decompress, needs_crc, needs_compress, needs_reencrypt, needs_decrypt, needs_encrypt;
     bool have_dos_time, dirent_changed;
     time_t mtime_before_copy;
+    have_dos_time = false;
 
     if (zip_source_stat(src, &st) < 0) {
         zip_error_set_from_source(&za->error, src);
