diff --git a/source/common/unicode/umachine.h b/source/common/unicode/umachine.h
index 09c887c80e..bd1e7b6ccd 100644
--- a/source/common/unicode/umachine.h
+++ b/source/common/unicode/umachine.h
@@ -290,7 +290,7 @@ typedef int8_t UBool;
 #   define U_DEFINE_FALSE_AND_TRUE 1
 #else
     // Outside ICU: Avoid collision with non-macro definitions of FALSE & TRUE.
-#   define U_DEFINE_FALSE_AND_TRUE 0
+#   define U_DEFINE_FALSE_AND_TRUE 1
 #endif
 
 #if U_DEFINE_FALSE_AND_TRUE || defined(U_IN_DOXYGEN)
