diff --git a/include/boost/detail/iterator.hpp b/include/boost/detail/iterator.hpp
index 95ec737..ba6a512 100644
--- a/include/boost/detail/iterator.hpp
+++ b/include/boost/detail/iterator.hpp
@@ -6,12 +6,6 @@
 #ifndef ITERATOR_DWA122600_HPP_
 #define ITERATOR_DWA122600_HPP_
 
-// This header is obsolete and deprecated.
-
-#include <boost/config/header_deprecated.hpp>
-
-BOOST_HEADER_DEPRECATED("<iterator>")
-
 #include <iterator>
 #if defined(__SUNPRO_CC) && (defined(__SGI_STL_PORT) || defined(_STLPORT_VERSION))
 #include <cstddef>
